cmd/main/main.go

```diff
@@ -1,38 +1,17 @@
 package main
 
 import (
-	"book_service/pkg/clients"
-	"book_service/pkg/routes"
+	app "book_service/pkg"
 	"book_service/pkg/utils"
-	"fmt"
-	"github.com/gin-gonic/gin"
-	"github.com/gin-gonic/gin/binding"
-	"github.com/joho/godotenv"
-	"log"
+	"github.com/sirupsen/logrus"
 )
 
 func main() {
-	err := godotenv.Load()
-	if err != nil {
-		fmt.Printf("Error loading .env file: %v", err)
-	}
+	app.SetupEnv()
+	app.InitClients()
```

Comment:
We should call here one setup function which is exported, and inside it, you should call all the needed initialization.

Should look something like this:
```suggestion
	config.Setup()
```

Reply:
For tests is convenient to separate the `app.SetupEnv()` from the InitClients etc, just for testing in isolations. Could be having another function for default "prod" setup could be valid, but I think its a bad pattern and hard to maintain if the prod version of the code is not the same as the dev or test code. (code smell)

Reply:
I didn't mean you can't have these 2 functions in separate, I meant that here you should call only `config.Setup()` (btw, you can name it app.Setup(), I just think that config is better).

Regarding the tests, you can test them separately inside config.

If tomorrow you want to add something to the config setup, the main() should still have the setup as a black box.